HARK/core.py

```diff
@@ -59,27 +250,23 @@ def get_parameter(self, name):
 
     def __eq__(self, other):
         if isinstance(other, type(self)):
-            return self.parameters == other.parameters
+            return self.parameters.to_dict() == other.parameters.to_dict()
 
         return NotImplemented
 
-    def __init__(self):
-        if not hasattr(self, "parameters"):
-            self.parameters = {}
-
     def __str__(self):
         type_ = type(self)
         module = type_.__module__
         qualname = type_.__qualname__
 
-        s = f"<{module}.{qualname} object at {hex(id(self))}.\n"
-        s += "Parameters:"
+        string = f"<{module}.{qualname} object at {hex(id(self))}.\n"
+        string += "Parameters:"
 
         for p in self.parameters:
-            s += f"\n{p}: {self.parameters[p]}"
+            string += f"\n{p}: {self.parameters[p]}"
```

Comment:
It would be better, while you're at it, for the Parameters object to have a `__repr__()` or `__str__()` method and to call that here.

https://www.digitalocean.com/community/tutorials/python-str-repr-functions

Reply:
will do